Document overridable YAML configuration (#118)
PaperMC/PaperDocs: source/server/configuration.rst

    Last updated Sep 3rd, 2021 for MC 1.17.1, Paper build #249

Overriding settings
===================

Paper introduced environment variables to override configuration values in `paper.yml`, `spigot.yml`, and `server.properties`.

> The environment variable name starts with the prefix (see the table below), following by the configuration key (``.`` and ``-`` is replaced with ``_``).

Reviewer: ```suggestion
The environment variable name starts with the prefix (see the table below), followed by the configuration key (``.`` and ``-`` are replaced with ``_``).
```

> Paper introduced environment variables to override configuration values in `paper.yml`, `spigot.yml`, and `server.properties`.
> The environment variable name starts with the prefix (see the table below), followed by the configuration key (``.`` and ``-`` are replaced with ``_``).
> This works with string, int, double, and boolean values.

Reviewer 2: I'm not really even sure if this is better, but something about how this is worded feels odd. Let me know what you think.

```suggestion
Paper has introduced environment variables to override configuration values in `paper.yml`, `spigot.yml`, and `server.properties`.
Each environment variable starts with a prefix (see the table below), followed by the configuration key, replacing ``.`` and ``-`` with ``_``.
This works with string, int, double, and boolean values.
```

    Last updated Sep 3rd, 2021 for MC 1.17.1, Paper build #249

Overriding settings
===================

Paper has introduced environment variables and JVM properties to override configuration values in `paper.yml`, `spigot.yml`, `bukkit.yml`, and `server.properties`.

Environment variables
~~~~~~~~~~~~~~~~~~~~~

Each environment variable starts with a prefix (see the table below), followed by the configuration key, replacing ``.`` and ``-`` with ``_``.

> This works with string, int, boolean, double, and long values.

Reviewer 3: Do you think this is enough to communicate that overriding may not be used on anything complex like maps or lists ?

Author: I think I may have to clarify it a bit more
